cpp/src/DataStorm/ConnectionManager.cpp

```diff
@@ -18,6 +18,10 @@ ConnectionManager::add(
     function<void(const ConnectionPtr&, exception_ptr)> callback)
 {
     lock_guard<mutex> lock(_mutex);
+
+    // Disable the inactivity timeout on the connection.
```

Comment:
I would do this before locking the mutex, because it's cleaner.

As written, you wonder if this locking is required or if there is a potential lock-acquisition issue between this mutex and the connection mutex.